# Input Fonksiyonu

Bu bölümde kullanıcıdan veri girişi almamıza olanak sağlayan *input()* fonksiyonunu öğreneceğiz.


### input() Fonkisyonu

Programımızın çalışması esnasında kullanıcıdan bilgi girişi yapmasını istediğimizde *input()* fonksiyonunu kullanabiliriz. Bu fonksiyon programımızın kullanıcı ile etkilişimli çalışmasına olanak sağlacaktır. 


In [1]:
input()

değer


'değer'

#### Kullacıdan Alınan Değerin Değişkene Atanması

Kullanıcı tarafından girilen bilgiler bir değişkene atanarak program içerisinde kullanımı sağlanabilir. Bunun için;

In [23]:
a = input()

Hello World


In [10]:
type(a)

str

Yukarıda kullanıcı tarafından girilen *Hello World* bilgisini *a* değişkeni içerisine atadık. *a* değişkenininin tipi kontrol edildiğinde *string* olduğu görülmektedir. 

*print()* fonksiyonu kullanarak ekrana yazdırmayı deneyelim;


In [9]:
print(a)

Hello World



#### *input()* ile Ekrana Veri Yazdırma

Kullanıcıdan veri girişi yapmasını isterken ona bir açıklama göstermek isteyebiliriz. Bu işlem için ek olarak *print()* kullanmamıza gerek yok. *input()* fonksiyonu içerisinde bunu yapabiliriz. 

In [11]:
company = input('Lütfen Firma İsmini Giriniz: ')

Lütfen Firma İsmini Giriniz: ACME


In [12]:
print(company)

ACME



#### *input()* ile Matematiksel İşlemler

*input()* fonkisyonu aracılığı ile kullanıcıdan alınan veriler *string* veri tipinde olacaktır. Matematiksel işlemler yapılacaksa bu durumun gözönün bulundurulması gerekmektedir.

In [13]:
glass_price = input('Lütfen Bardak Birim Fiyatını Giriniz: ')

Lütfen Bardak Birim Fiyatını Giriniz: 55


In [14]:
type(glass_price)

str

In [22]:
total_price = int(glass_price)

bill_text = """
Toplam Tutarınız:
                 {0} TL

"""

print(bill_text.format(total_price))


Toplam Tutarınız:
                 55 TL





Başka bir örnek yapalım;

In [35]:
current_wallet = 1.5
cash_amount = input("Yüklemek istediğiniz tutarı giriniz: ")

current_wallet = current_wallet + int(cash_amount)

Yüklemek istediğiniz tutarı giriniz: 22.5


ValueError: invalid literal for int() with base 10: '22.5'



Kullanıcı girişi olarak *22.5* girişi yapıldı. Bu bir *float* değer. Dönüşüm de hata yaptık. Kodumuzu düzenliyoruz;


In [36]:
current_wallet = 1.5
cash_amount = input("Yüklemek istediğiniz tutarı giriniz: ")

current_wallet += float(cash_amount)

bill_text = """
Güncel Bakiyeniz:
                 {0} EU

"""


print(bill_text.format(current_wallet))

Yüklemek istediğiniz tutarı giriniz: 22.5

Güncel Bakiyeniz:
                 24.0 EU


